packages/scene-composer/src/components/toolbars/floatingToolbar/FloatingToolbar.tsx

```diff
@@ -33,17 +36,52 @@ export function FloatingToolbar({
 }: FloatingToolbarProps) {
   const sceneComposerId = useContext(sceneComposerIdContext);
   const { addingWidget } = useEditorState(sceneComposerId);
+  const [toolbarOrientation, setToolbarOrientation] = useState<ToolbarOrientation>(ToolbarOrientation.Vertical);
+  const [toolbarHeightPx, setToolbarHeightPx] = useState(0);
+  const [canvasHeightPx, setCanvasHeightPx] = useState<number | null>(null);
+  const canvas = document.getElementById('unselectable-canvas') as HTMLCanvasElement | null;
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (canvas && canvas.clientHeight) setCanvasHeightPx(canvas.clientHeight);
+    };
+
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [canvas?.clientHeight]);
+
+  useEffect(() => {
```

Comment:
You can remove this `useEffect` and call `handleResize()` at line# 50.

Reply:
Good point, fixed